packages/components/src/disabled/index.js

```diff
@@ -79,22 +55,43 @@ class Disabled extends Component {
 				focusable.setAttribute( 'contenteditable', 'false' );
 			}
 		} );
-	}
+	};
+
+	// Debounce re-disable since disabling process itself will incur
+	// additional mutations which should be ignored.
+	const debouncedDisable = useCallback(
+		debounce( disable, { leading: true } ),
+		[]
+	);
+
+	useLayoutEffect( () => {
+		disable();
+
+		const observer = new window.MutationObserver( debouncedDisable );
+		observer.observe( node.current, {
+			childList: true,
+			attributes: true,
+			subtree: true,
+		} );
 
-	render() {
-		const { className, ...props } = this.props;
-		return (
-			<Provider value={ true }>
-				<div
-					ref={ this.bindNode }
-					className={ classnames( className, 'components-disabled' ) }
-					{ ...props }
-				>
-					{ this.props.children }
-				</div>
-			</Provider>
-		);
-	}
+		return () => {
+			observer.disconnect();
+			debouncedDisable.cancel();
+		};
+	}, [] );
+
+	//Lets return
```

Comment:
Seems like a useless comment?

Reply:
Lol. I cannot remember why I ever wrote that comment.